### DISCUSSION TITLE
Rotary encoder unreliable (on Mega/RAMPS 1.4/Reprapdiscount full smart controller)

### DISCUSSION THREAD

**Discussion by mvackel** [Paciente8159/uCNC#497] [Config and issues] [answered]
The rotary encoder does not show a consistent response: about 30% of the clockwise rotation is being interpreted wrongly as being counter-clockwise and vice-versa.

This is most noticeable when jogging the axes. Trying to move, say 10mm to the right requires many attempts with the motors going many times in the opposite desired direction.

Is there a way to make it more consistent?

**Comment by Paciente8159** [comment]
I got to see if adding a few short delays on input read to try to capture the next state right away fixes that.
The encoder is connected to non interruptable pin so coupling that to the encoder module is not possible.

**Comment by Paciente8159** [comment]
[Modified condition states for encoder rotation detection](https://github.com/Paciente8159/uCNC-modules/commit/ee8939d583139c6871e5fe59cfcbc2fd62a50c19).
Also fixed the idle screen for 5 or more axis (missing divisor line on the last axis).
It seems more responsive. It's still a bit imprecise (detects wrong direction) but I think I'm at the limits of what I can do via soft polling.
Coupled with #501 jogging should be more controllable.

Another way to improve jog experience is to increase the jog distance for each command (for example to 10mm). Pressing the button issues a jog cancel command and stops the jog motion.

**Comment by Paciente8159** [comment, marked as answer]
Please download and try this modified version of the graphic_display in this branch.

https://github.com/Paciente8159/uCNC-modules/tree/modified-encoder-detect

This adds a sampler and accumulator as well as a debounce to the encoder button.
In my tests this vastly improved the encoder reliability at the expense of some responsiveness, but not at the point that it becomes unusable.

**Comment by mvackel** [comment]
It's now much better an totally functional. I think you solved the problem. Great!
